packages/0xcert-ethereum-utils-contracts/src/tests/bytes-to-types.test.ts

```diff
@@ -0,0 +1,34 @@
+import { Spec } from '@specron/spec';
+
+interface Data {
+  bytesToTypes?: any;
+}
+
+const spec = new Spec<Data>();
+
+spec.beforeEach(async (ctx) => {
+  const bytesToTypes = await ctx.deploy({
+    src: './build/unpack-test-mock.json',
+    contract: 'UnpackTestMock',
+  });
+  ctx.set('bytesToTypes', bytesToTypes);
+});
+
+spec.test('correctly unpacks bytes to types', async (ctx) => {
```

Comment:
Can we please get a gas test here to verify the hypothesis that scanning backwards is cheaper than scanning forwards?

One test to show gas cost scanning forward. And one test scanning backwards.

If they are the same then I would design `_doActionsReplaceZeroAddress` a little differently and simpler.

Reply:
I have been researching a little and came across this:
https://www.dhondt.tech/blog/2019/1/Parse-bytes-argument-in-Solidity.html and https://modex.tech/developers/idiana96/solidity-bytes-utils/src/master/contracts/BytesLib.sol

If I understand this correctly `mload` always reads 32bytes. So reading address that has 20 bytes in a packed encoding cannot be done with `mload` but parsing backwards it will not have 32 bytes available and will reads what is available.

Reply:
@fulldecent comments?

Reply:
I believe Solidity is already masking this in the `return`. Would like to see the test.